### ISSUE TITLE
Bounding Boxes are not aligned and appear not in the picture

### ISSUE THREAD

**Issue by mojoee** [Microsoft/singleshotpose#139]
Does anyone have experienced problems with the bounding boxes in the visual presentation? I have basically 2 kind of problems after training as shown in the picture: 1)Bounding Box (GT and PR) are not aligned with the real object and 2) bounding boxes appear outside of the image. Can anyone help me ?
![image](https://user-images.githubusercontent.com/47809126/75123945-c07b6300-56e6-11ea-949e-d102735ff786.png)

**Comment by super-zyw** [comment]
Do you have enough training data?

**Comment by capslocknanda** [comment]
can you help me with how to visualize the prediction on an image ?

**Comment by mojoee** [comment]
the code for the visualization is in the jupyter notebook file called valid.ipynb. You just need to run this file and you will get the visualizations.

**Comment by siatlpl** [comment]
I have the same problems, do you have solve it? thank you

**Comment by nithinvenny07** [comment]
I am also facing same problem did anyone found solution??

**Comment by Allwinraj** [comment]
I am also facing the same issue. for my own data.
The Groudtruth also shows the wrong location while in training and testing the model. but the loss is at a minimum range.
Anyone had a solution??

**Comment by Barry-Liang** [comment]
I have the same problem for my own data. My case is a single object detection with classNum=1. Both training and validation work well. But after that, I use the camera to do the real time detection, even when there is no object in the image, the bounding box will appear. In addition, the bounding box can become very large or in wrong position.

**Comment by mojoee** [comment]
I remember I had to change the intrinsic camera parameters! Does that work?

**Comment by Barry-Liang** [comment]
Yes, I used the intern realsence camera to collect data and train the network. The validation works well with reasonable accuracy. After that, I use another RGB camera to do the realtime detection. I have changed the intrinsic parameters in the corresponding <object>.data files. And then the errors occur which are just the same as your problem.

**Comment by mojoee** [comment]
I can't recall how exactly I solved it. And it's been too long. Hope someone else can help

**Comment by super-zyw** [comment]
I am doing all good till now, what is the exact issue here?

**Comment by nithinvenny07** [comment]
@mojoee may i know how u have created the labels for the dataset thanks in advance
